### PR TITLE
[CORE] Fix non-deterministic filter executed twice when push down to scan

### PR DESCRIPTION
**Pull request by zml1206**
## What changes were proposed in this pull request?

 `PredicateHelper.getRemainingFilters` has a bug, `ExpressionSet` can only remove deterministic expression, it will cause non-deterministic filter to be executed twice.

## How was this patch tested?

UT.

### CODE REVIEW

File: backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala [hunk 1]
```diff
@@ -1892,4 +1892,18 @@ class TestOperator extends VeloxWholeStageTransformerSuite with AdaptiveSparkPla
...
+    val scans = plan.collect { case scan: FileSourceScanExecTransformer => scan }
+    val filters = plan.collect { case filter: FilterExecTransformer => filter }
+    assert(scans.size == 1)
+    assert(filters.size == 1)
+    assert(scans(0).dataFilters.size == 1)
+    val remainingFilters = FilterHandler.getRemainingFilters(
+      scans(0).dataFilters,
+      splitConjunctivePredicates(filters(0).condition))
+    assert(remainingFilters.size == 0)
+  }
```
[the remark this answers is not in the document]
Comment (zml1206) [reply backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala:1908]: Result length check in #6191 is not absolutely stable, so I used the method test.

[the remark this answers is not in the document]
Comment (zhztheplayer) [reply backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala:1908]: It's stable as the probability of false positive is nearly zero

https://www.wolframalpha.com/input?i=P%5BX+%3E+25000%5D+for+X%7EB%2860000%2C0.25%29

You can use 25000 < x < 35000 which is very much enough

**Comment by zml1206** [comment]
Run Gluten Clickhouse CI

**Comment by zml1206** [comment]
The error does not seem to be related to this PR. cc @zhztheplayer

**Review by zhztheplayer — APPROVED** [review]

**Pull request merged**